Userland/Games/GameOfLife/Pattern.cpp

```diff
@@ -24,13 +23,13 @@ void Pattern::set_action(GUI::Action* action)
 
 void Pattern::rotate_clockwise()
 {
-    Vector<DeprecatedString> rotated;
+    Vector<StringView> rotated;
     for (size_t i = 0; i < m_pattern.first().length(); i++) {
         StringBuilder builder;
         for (int j = m_pattern.size() - 1; j >= 0; j--) {
-            builder.append(m_pattern.at(j).substring(i, 1));
+            builder.append(m_pattern.at(j).substring_view(i, 1));
         }
-        rotated.append(builder.to_deprecated_string());
+        rotated.append(builder.string_view());
     }
     m_pattern = move(rotated);
```

Comment:
This smells like it'll cause a UAF

Reply:
You're absolutely right :facepalm:. I've modified the PR to construct `Pattern` objects from `String` instead and made sure that rotating actually works this time!